### PR TITLE
[KEYCLOAK-11764] Upgrade to Wildfly 19

### PR DESCRIPTION
**Pull request by douglaspalmer**

### CODE REVIEW

**Comment by stianst** [comment]
@hmlnarik since your team has done the last few upgrades, please make sure someone from your team reviews the upgrade to WF 19

**Comment by hmlnarik** [comment]
@vramik @martin-kanis Could you please keep an eye on this PR?

**Review by abstractj — COMMENTED** [review]
@douglaspalmer adapter tests are failing and it seems related with the upgrade:

```
- [ERROR] Failures: 
-- [ERROR]   SAMLLoginResponseHandlingTest.testErrorHandlingSigned:142->lambda$testErrorHandlingSigned$6:142 
-- Expected: response body matches a string containing "ERROR_STATUS"
--      but: was <HttpResponseProxy{HTTP/1.1 200 OK [Connection: keep-alive, Content-Type: text/html;charset=UTF-8, Content-Length: 117, Date: Wed, 25 Mar 2020 00:43:12 GMT] ResponseEntityProxy{[Content-Type: text/html;charset=UTF-8,Content-Length: 117,Chunked: false]}}> with entity <html><head><title>Error Page</title></head><body><h1>There was an error</h1><br/>HTTP status code: 403</body></html>
-- [ERROR]   SAMLLoginResponseHandlingTest.testErrorHandlingUnsigned:128->lambda$testErrorHandlingUnsigned$4:128 
-- Expected: response body matches a string containing "INVALID_SIGNATURE"
--      but: was <HttpResponseProxy{HTTP/1.1 200 OK [Connection: keep-alive, Content-Type: text/html;charset=UTF-8, Content-Length: 117, Date: Wed, 25 Mar 2020 00:43:12 GMT] ResponseEntityProxy{[Content-Type: text/html;charset=UTF-8,Content-Length: 117,Chunked: false]}}> with entity <html><head><title>Error Page</title></head><body><h1>There was an error</h1><br/>HTTP status code: 403</body></html>
-- [INFO] 
```

**Comment by vramik** [comment]
@douglaspalmer there is also jboss-parent 34 in keycloak-bom-parent. It should be updated to 35 as well.

**Comment by douglaspalmer** [comment]
> @douglaspalmer there is also jboss-parent 34 in keycloak-bom-parent. It should be updated to 35 as well.

Fixed, thanks.

**Comment by martin-kanis** [comment]
Am I the only one who is getting a compilation error? 

Could not resolve dependencies for project org.keycloak:keycloak-saml-core-public:jar:9.0.4-SNAPSHOT: Could not find artifact org.slf4j:slf4j-api:jar:1.7.22.jbossorg-1 in central (https://repo.maven.apache.org/maven2)

I saw this in previous upgrade and reason was that org.slf4j:slf4j-api:jar:1.7.22.jbossorg-1 is only in jboss nexus. In the end we kept our old version.

File: pom.xml [hunk 1]
```diff
@@ -41,35 +41,35 @@
...
-        <cxf.undertow.version>3.3.4</cxf.undertow.version>
+        <bouncycastle.version>1.62</bouncycastle.version>
+        <cxf.version>3.3.5</cxf.version>
+        <cxf.jetty.version>3.3.5</cxf.jetty.version>
+        <cxf.jaxrs.version>3.3.5</cxf.jaxrs.version>
+        <cxf.undertow.version>3.3.5</cxf.undertow.version>
         <dom4j.version>2.1.1</dom4j.version>
-        <github.relaxng.version>2.3.1</github.relaxng.version>
-        <h2.version>1.4.193</h2.version>
+        <github.relaxng.version>2.3.3-b02</github.relaxng.version>
```
Comment (martin-kanis) [inline pom.xml]: I just noticed this dependency has a different GAV than in Wildfly. It doesn't even exist in this version. Please remove it completely as it's not used.

File: pom.xml [hunk 2]
```diff
@@ -41,35 +41,35 @@
...
         <jakarta.persistence.version>2.2.3</jakarta.persistence.version>
-        <hibernate.core.version>5.3.14.Final</hibernate.core.version>
-        <hibernate.c3p0.version>5.3.14.Final</hibernate.c3p0.version>
-        <infinispan.version>9.4.16.Final</infinispan.version>
+        <hibernate.core.version>5.3.15.Final</hibernate.core.version>
+        <hibernate.c3p0.version>5.3.15.Final</hibernate.c3p0.version>
+        <infinispan.version>9.4.18.Final</infinispan.version>
         <!-- Will be used in the product. Upstream versions are overridden in the community profile -->
-        <jackson.version>2.10.0</jackson.version>
+        <jackson.version>2.10.1</jackson.version>
```
Comment (martin-kanis) [inline pom.xml]: This version is used for product. Community version is down in community profile. Later when EAP CD19 will be out and the versions will be the same, we can simplify this.

File: pom.xml [hunk 3]
```diff
@@ -111,8 +111,9 @@
         <!-- Others -->
         <commons-lang.version>2.6</commons-lang.version>
         <commons-lang3.version>3.9</commons-lang3.version>
-        <apacheds.version>2.0.0-M21</apacheds.version>
-        <apacheds.codec.version>1.0.0-M33</apacheds.codec.version>
+        <commons-io.version>2.6</commons-io.version>
+        <apacheds.version>2.0.0-M24</apacheds.version>
+        <apacheds.codec.version>1.0.0</apacheds.codec.version>
```
Comment (martin-kanis) [inline pom.xml]: I didn't find this GAV in Wildfly.

**Review by martin-kanis — CHANGES_REQUESTED** [review]
Also licenses should be updated:
`distribution/adapters/as7-eap6-adapter/as7-modules/src/main/resources/licenses/keycloak/licenses.xml`
`distribution/adapters/fuse-adapter-zip/src/main/resources/licenses/keycloak/licenses.xml`
`distribution/feature-packs/server-feature-pack/src/main/resources/licenses/keycloak/licenses.xml`
and then run `./distribution/licenses-common/update-licenses.sh` script

**Comment by abstractj** [comment]
> Am I the only one who is getting a compilation error?
> 
> Could not resolve dependencies for project org.keycloak:keycloak-saml-core-public:jar:9.0.4-SNAPSHOT: Could not find artifact org.slf4j:slf4j-api:jar:1.7.22.jbossorg-1 in central (https://repo.maven.apache.org/maven2)
> 
> I saw this in previous upgrade and reason was that org.slf4j:slf4j-api:jar:1.7.22.jbossorg-1 is only in jboss nexus. In the end we kept our old version.

@martin-kanis I don't think so, I got the same here in a fresh environment. @douglaspalmer could you please give it a try?

**Comment by douglaspalmer** [comment]
> Am I the only one who is getting a compilation error?
> 
> Could not resolve dependencies for project org.keycloak:keycloak-saml-core-public:jar:9.0.4-SNAPSHOT: Could not find artifact org.slf4j:slf4j-api:jar:1.7.22.jbossorg-1 in central (https://repo.maven.apache.org/maven2)
> 
> I saw this in previous upgrade and reason was that org.slf4j:slf4j-api:jar:1.7.22.jbossorg-1 is only in jboss nexus. In the end we kept our old version.

You need to add jboss nexus to your settings file. Alternatively, you can run mvn -s maven-settings.xml and use the settings file we provide. Surely it's better to use the same versions wildly uses?

File: pom.xml (hunk 3 above)
Comment (douglaspalmer) [reply pom.xml]: mvn dependency:list on wildfly didn't list a version for the codec. I selected the version for codec from the parent pom for apacheds 2.0.0-M24.

File: pom.xml (hunk 1 above)
Comment (douglaspalmer) [reply pom.xml]: Done

File: pom.xml (hunk 2 above)
Comment (douglaspalmer) [reply pom.xml]: Done

**Comment by martin-kanis** [comment]
> > Am I the only one who is getting a compilation error?
> > Could not resolve dependencies for project org.keycloak:keycloak-saml-core-public:jar:9.0.4-SNAPSHOT: Could not find artifact org.slf4j:slf4j-api:jar:1.7.22.jbossorg-1 in central (https://repo.maven.apache.org/maven2)
> > I saw this in previous upgrade and reason was that org.slf4j:slf4j-api:jar:1.7.22.jbossorg-1 is only in jboss nexus. In the end we kept our old version.
> 
> You need to add jboss nexus to your settings file. Alternatively, you can run mvn -s maven-settings.xml and use the settings file we provide. Surely it's better to use the same versions wildly uses?

We discussed this with Stian on keycloak-team mailing list. Search for "Can't build distribution from master" email from Oct 9th 2019. He said, that we should use slf4j version from maven central as we don't include slf4j ourselves in our modules but only as module's dependency from wildfly/eap.

**Comment by douglaspalmer** [comment]
> > > Am I the only one who is getting a compilation error?
> > > Could not resolve dependencies for project org.keycloak:keycloak-saml-core-public:jar:9.0.4-SNAPSHOT: Could not find artifact org.slf4j:slf4j-api:jar:1.7.22.jbossorg-1 in central (https://repo.maven.apache.org/maven2)
> > > I saw this in previous upgrade and reason was that org.slf4j:slf4j-api:jar:1.7.22.jbossorg-1 is only in jboss nexus. In the end we kept our old version.
> > 
> > 
> > You need to add jboss nexus to your settings file. Alternatively, you can run mvn -s maven-settings.xml and use the settings file we provide. Surely it's better to use the same versions wildly uses?
> 
> We discussed this with Stian on keycloak-team mailing list. Search for "Can't build distribution from master" email from Oct 9th 2019. He said, that we should use slf4j version from maven central as we don't include slf4j ourselves in our modules but only as module's dependency from wildfly/eap.

Ok, I've reverted the change.

**Review by martin-kanis — CHANGES_REQUESTED** [review]
@douglaspalmer Thanks for the changes. I have few more comments (see inline). 

Update one forgotten Keycloak license file, please
```
distribution/saml-adapters/as7-eap6-adapter/as7-modules/src/main/resources/licenses/keycloak/licenses.xml
```
then run again 
```
distribution/licenses-common/update-licenses.sh
```

### **UPDATE**
Forgot about EAP 7.4.0.CD19 upgrade, the product build is failing https://issues.redhat.com/browse/KEYCLOAK-13636.

**Comment by douglaspalmer** [comment]
> @douglaspalmer Thanks for the changes. I have few more comments (see inline).
> 
> Update one forgotten Keycloak license file, please
> 
> ```
> distribution/saml-adapters/as7-eap6-adapter/as7-modules/src/main/resources/licenses/keycloak/licenses.xml
> ```
> 
> then run again
> 
> ```
> distribution/licenses-common/update-licenses.sh
> ```
> 

This is already included, does it need further changes?

**Comment by martin-kanis** [comment]
> > @douglaspalmer Thanks for the changes. I have few more comments (see inline).
> > Update one forgotten Keycloak license file, please
> > ```
> > distribution/saml-adapters/as7-eap6-adapter/as7-modules/src/main/resources/licenses/keycloak/licenses.xml
> > ```
> > 
> > 
> > then run again
> > ```
> > distribution/licenses-common/update-licenses.sh
> > ```
> 
> This is already included, does it need further changes?

This is eap6 **saml** adapter licenses. You changed only eap6 **oidc** adapter licenses.

File: pom.xml [hunk 4]
```diff
@@ -41,35 +41,33 @@
 
         <product.build-time>${timestamp}</product.build-time>
 
-        <wildfly.version>18.0.1.Final</wildfly.version>
-        <wildfly.build-tools.version>1.2.12.Final</wildfly.build-tools.version>
-        <eap.version>7.3.0.GA-redhat-00004</eap.version>
-        <wildfly.core.version>10.0.3.Final</wildfly.core.version>
+        <wildfly.version>19.0.0.Final</wildfly.version>
+        <wildfly.build-tools.version>1.2.13.Final</wildfly.build-tools.version>
+        <eap.version>7.4.0.CD19-redhat-00001</eap.version>
```
Comment (martin-kanis) [inline pom.xml]: Please keep old version 7.3.0.GA-redhat-00004 for now. Build with new one is broken in PNC. Sorry for confusion.

Comment (douglaspalmer) [reply pom.xml]: Ok, reverted

**Review by martin-kanis — DISMISSED** [review]

**Review by vramik — CHANGES_REQUESTED** [review]
Current pipeline doesn't work with the PR, due to upgrade of jboss-parent to 35, all http repositories are banned. There are some http repositories used there so another PR to pipeline is needed.

https://keycloak-jenkins.com/job/universal-test-pipeline-server/1611/execution/node/1491/log/

EDIT: maybe we can switch to download-maven-plugin for those dependencies which require the fuse repo, similar to https://github.com/keycloak/keycloak/blob/master/testsuite/integration-arquillian/servers/cache-server/jboss/infinispan/pom.xml#L50-L67

**Comment by abstractj** [comment]
@douglaspalmer could you please follow up on @vramik's comments?

**Comment by douglaspalmer** [comment]
@vramik maybe we should run the pipeline with -Denforcer.skip? The indy repository doesn't appear to be available on https. Neither is the fuse nexus.

**Comment by douglaspalmer** [comment]
@vramik alternatively, I can disable that enforcer rule in the Keycloak parent pom.

**Comment by vramik** [comment]
@douglaspalmer running pipeline with -Denforcer.skip skipps all enforcing rules not just the https one. So IMO this is not something what we want. I'm not sure about disabling the rule in parent pom, this should be discussed in wider audience, perhaps keycloak-dev? Or maybe keycloak-team mailing list.

**Review by slaskawi — DISMISSED** [review]

**Comment by vramik** [comment]
@douglaspalmer what's the status on the `-Dinsecure.repositories=WARN` for the pipeline? I believe it's the only thing blocking the PR to be merged.

**Comment by douglaspalmer** [comment]
> @douglaspalmer what's the status on the `-Dinsecure.repositories=WARN` for the pipeline? I believe it's the only thing blocking the PR to be merged.

@vramik https://github.com/keycloak/keycloak-pipeline/pull/248

**Comment by martin-kanis** [comment]
I have two follow up commits here https://github.com/martin-kanis/keycloak/commits/cd19. First is alignment to CD19 and second one fix of KEYCLOAK-13636. Do you want to include those commits in this PR?

**Comment by vramik** [comment]
@douglaspalmer thanks. There are few failed tests in the pipeline run with this PR, but those might be unrelated to the changes. Maybe we could rebase it to latest upstream to be sure.

**Comment by douglaspalmer** [comment]
> I have two follow up commits here https://github.com/martin-kanis/keycloak/commits/cd19. First is alignment to CD19 and second one fix of KEYCLOAK-13636. Do you want to include those commits in this PR?

@martin-kanis I'd rather the CD19 alignment was done as a separate JIRA given that those changes are for EAP and this PR is for Wildfly.

**Review by vramik — CHANGES_REQUESTED** [review]
@douglaspalmer thanks, rebase solved all issues in server pipeline. Regarding adapter pipeline there are still some changes needed.

Springboot tests unfortunately uses mvn call from process-exec-maven-plugin:

- [integration-arquillian-tests-springboot/pom.xml#L70](https://github.com/keycloak/keycloak/blob/master/testsuite/integration-arquillian/tests/other/springboot-tests/pom.xml#L70)
- [integration-arquillian-tests-springboot/pom.xml#L92](https://github.com/keycloak/keycloak/blob/master/testsuite/integration-arquillian/tests/other/springboot-tests/pom.xml#L92)
- [integration-arquillian-tests-springboot/pom.xml#L148](https://github.com/keycloak/keycloak/blob/master/testsuite/integration-arquillian/tests/other/springboot-tests/pom.xml#L148)
- [integration-arquillian-tests-springboot/pom.xml#L173](https://github.com/keycloak/keycloak/blob/master/testsuite/integration-arquillian/tests/other/springboot-tests/pom.xml#L173)

due to that fact the rule from jboss-parent fails: https://keycloak-jenkins.com//job/universal-test-pipeline-adapters/1344/execution/node/1336/log/?consoleFull

All other adapters are ok, so this should be the last necessary change.

**Review by martin-kanis — APPROVED** [review]

**Review by vramik — APPROVED** [review]
@douglaspalmer thanks

**Review by abstractj — APPROVED** [review]

**Pull request merged**